nuskybgd/model.py

```diff
@@ -523,71 +523,138 @@ def applymodel_intbgd(presets, refspec, bgddetimsum, model_num, src_number=None,
+#             m.apec.Abundanc.frozen = True
+#             m.apec.Redshift.frozen = True
+#             m.apec.norm.values = np.sum(
+#                 bgddetimsum[spec_arrinx] * np.array(
+#                     mod_intbgd['apec'][fpm]['ifactors'])
+#             ) / np.sum(bgddetimsum[spec_arrinx])
 
-            m.apec.kT.values = mod_intbgd['apec'][fpm]['kt']
-            m.apec.Abundanc.values = mod_intbgd['apec'][fpm]['abundanc']
-            m.apec.Redshift.values = mod_intbgd['apec'][fpm]['redshift']
-            m.apec.kT.frozen = True
-            m.apec.Abundanc.frozen = True
-            m.apec.Redshift.frozen = True
-            m.apec.norm.values = np.sum(
-                bgddetimsum[spec_arrinx] * np.array(
-                    mod_intbgd['apec'][fpm]['ifactors'])
-            ) / np.sum(bgddetimsum[spec_arrinx])
 
             # 3 solar lines and 19 keV line -- load them from preset
+            ref_set = False
             for attr in ['lorentz', 'lorentz_3', 'lorentz_4', 'lorentz_5']:
-                m_line = getattr(m, attr)
-                m_line.LineE.values = mod_intbgd[attr][fpm]['linee']
-                m_line.Width.values = mod_intbgd[attr][fpm]['width']
-                m_line.LineE.frozen = True
-                m_line.Width.frozen = True
-                m_line.norm.values = np.sum(
+                line_norm = np.sum(
                     bgddetimsum[spec_arrinx] * np.array(
                         mod_intbgd[attr][fpm]['ifactors'])
                 ) / np.sum(bgddetimsum[spec_arrinx])
+                
+                if not ref_set:
+                    ref_set = True
+                    ref_norm = line_norm
+                    ref_par = int(parnum+2) + m_ref_npar_offset
+                    this_par = {parnum:f"{mod_intbgd[attr][fpm]['linee']}, -0.1",
+                            parnum+1:f"{mod_intbgd[attr][fpm]['width']}, -0.1",
+                            parnum+2:f"{line_norm}, 0.1"}
+                else:
+                    rel_norm = line_norm / ref_norm                
+                    this_par = {parnum:f"{mod_intbgd[attr][fpm]['linee']}, -0.1",
+                                parnum+1:f"{mod_intbgd[attr][fpm]['width']}, -0.1",
+                                parnum+2:f"={model_name}:{ref_par}*{rel_norm}"}
+                par_dict.update(this_par)
+                parnum += 3
+
+                continue
+                # Below is the old way to do this
+#                 m_line = getattr(m, attr)
+#                 m_line.LineE.values = mod_intbgd[attr][fpm]['linee']
+#                 m_line.Width.values = mod_intbgd[attr][fpm]['width']
+#                 m_line.LineE.frozen = True
+#                 m_line.Width.frozen = True
+#                 m_line.norm.values = np.sum(
+#                     bgddetimsum[spec_arrinx] * np.array(
+#                         mod_intbgd[attr][fpm]['ifactors'])
+#                 ) / np.sum(bgddetimsum[spec_arrinx])
+                
```

Comment:
My understanding is that this is how the IDL program behaves, which lets these three solar related lines to be free (from nuskybgd_fitab.pro, ref below).

https://github.com/NuSTAR/nuskybgd/blob/9312caf14215eb1f29987f5fe1f9722b7b2d63d8/pro/nuskybgd_fitab.pro#L278